### PR TITLE
use patch instead of replace

### PR DESCRIPTION
**Pull request by Renkai**
Signed-off-by: Renkai <gaelookair@gmail.com>

###  What have you changed?
Use patch instead of replace in `Cargo.toml` & update Prometheus dependency to use the crate-io version.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- Unit test

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by breeswish** [comment]
@BusyJay PTAL

File: Cargo.toml [hunk 1]
```diff
@@ -151,14 +150,14 @@ test_sst_importer = { path = "components/test_sst_importer" }
...
-[replace]
+[patch.crates-io]
 # TODO: remove this when new raft-rs is published.
-"raft:0.6.0-alpha" = { git = "https://github.com/pingcap/raft-rs", default-features = false }
-"raft-proto:0.6.0-alpha" = { git = "https://github.com/pingcap/raft-rs", default-features = false }
-"protobuf:2.8.0" = { git = "https://github.com/pingcap/rust-protobuf", rev="b67d432c1b74350b38a5d96ddf885ac6c3ff46f5" }
-"protobuf-codegen:2.8.0" = { git = "https://github.com/pingcap/rust-protobuf", rev="b67d432c1b74350b38a5d96ddf885ac6c3ff46f5" }
-"prost:0.5.0" = { git = "https://github.com/danburkert/prost", rev = "1944c27c3029d01ff216e7b126d846b6cf8c7d77" }
-"protobuf-build:0.10.0" = { git = "https://github.com/tikv/protobuf-build", rev="1fe5d77689d9691b7917215f3a1d32968115cc02" }
+raft = { git = "https://github.com/pingcap/raft-rs", branch = "master" }
```
Comment (BusyJay) [inline Cargo.toml]: So how is features handled in patch?

File: components/pd_client/Cargo.toml [hunk 2]
```diff
@@ -34,6 +34,5 @@ tokio-timer = "0.2"
 txn_types = { path = "../txn_types" }
 
 [dependencies.prometheus]
-git = "https://github.com/tikv/rust-prometheus.git"
-rev = "d919ccd35976b9b84b8d03c07138c1cc05a36087"
+version = "0.8"
```
Comment (BusyJay) [inline components/pd_client/Cargo.toml]: You can write "prometheus = "0.8" now just like other dependencies.

File: Cargo.toml (hunk 1 above)
Comment (sticnarf) [reply Cargo.toml]: `[patch]` uses the same feature settings with the original one.

**Comment by Renkai** [comment]
@BusyJay PTAL

File: Cargo.toml [hunk 3]
```diff
@@ -133,10 +133,7 @@ uuid = { version = "0.7", features = ["serde", "v4"] }
 vlog = "0.1.4"
 into_other = { path = "components/into_other" }
```
Comment (BusyJay) [inline Cargo.toml]: Redundant line.

File: components/backup/Cargo.toml [hunk 4]
```diff
@@ -73,8 +73,7 @@ tokio-threadpool = "0.1"
 txn_types = { path = "../txn_types" }
 
 [dependencies.prometheus]
-git = "https://github.com/tikv/rust-prometheus.git"
-rev = "d919ccd35976b9b84b8d03c07138c1cc05a36087"
+version = "0.8"
```
Comment (BusyJay) [inline components/backup/Cargo.toml]: This can be rewritten too.

File: components/sst_importer/Cargo.toml [hunk 5]
```diff
@@ -44,8 +44,7 @@ txn_types = { path = "../txn_types" }
 uuid = { version = "0.7", features = ["serde", "v4"] }
 
 [dependencies.prometheus]
-git = "https://github.com/tikv/rust-prometheus.git"
-rev = "d919ccd35976b9b84b8d03c07138c1cc05a36087"
+version = "0.8"
```
Comment (BusyJay) [inline components/sst_importer/Cargo.toml]: This can be rewritten too.

File: cmd/Cargo.toml [hunk 6]
```diff
@@ -82,10 +82,7 @@ txn_types = { path = "../components/txn_types" }
 vlog = "0.1.4"
 cdc = { path = "../components/cdc" }
```
Comment (BusyJay) [inline cmd/Cargo.toml]: Seems this file is missed.

File: components/engine/Cargo.toml [hunk 7]
```diff
@@ -37,10 +37,7 @@ tikv_alloc = { path = "../tikv_alloc" }
 tikv_util = { path = "../tikv_util" }
 time = "0.1"
```
Comment (BusyJay) [inline components/engine/Cargo.toml]: Ditto.

**Comment by Renkai** [comment]
@sre-bot /test

**Comment by Renkai** [comment]
@sre-bot /merge

**Pull request merged**